crates/bifrost/src/loglets/local_loglet/log_store.rs

```diff
@@ -106,12 +111,17 @@ impl RocksDbLogStore {
     }
 }
 
-fn db_options() -> rocksdb::Options {
+fn db_options(options: &LocalLogletOptions) -> rocksdb::Options {
     let mut opts = rocksdb::Options::default();
     //
     // no need to retain 1000 log files by default.
     //
     opts.set_keep_log_file_num(10);
+
+    if !options.rocksdb.rocksdb_disable_wal() {
+        opts.set_manual_wal_flush(options.batch_wal_flushes);
+    }
```

Comment:
hmm, okay then.

Reply:
Do I understand it correctly to leave it as it is?

Reply:
What you mentioned calls for enabling atomic-flushes unconditionally (that's not true atm IIRC).

Reply:
Yes, at the moment we don't enable atomic flushes. Only the partition processor storage does it as part of this PR since it does not enable WAL by default.

The underlying problem seems to be that some options which depend on other options (e.g. atomic flushing) need to be set at opening time of the RocksDB instance whereas WAL can be disabled/enabled for every write.

Reply:
We can enable atomic flushes unconditionally to be on the safe side. What I am not sure is what the costs of atomic flushes are when WAL is enabled therefore the former is not needed.